### PR TITLE
Sentry Grenade now has a timer

### PR DESCRIPTION
**Pull request by DiamondWalker**
also includes new constructor variants for controlling explosion time

### CODE REVIEW

File: src/main/java/net/journey/entity/projectile/EntitySentryKingGrenade.java [hunk 1]
```diff
@@ -29,25 +30,74 @@ public boolean apply(@Nullable Entity e) {
 	public EntitySentryKingGrenade(World worldIn) {
 		super(worldIn);
 		this.pickupStatus = PickupStatus.DISALLOWED;
+		this.grenadeTimer = 200;
+
+	}
+	
+	public EntitySentryKingGrenade(World worldIn, int grenadeTime) {
```
Comment (TimeConqueror) [inline src/main/java/net/journey/entity/projectile/EntitySentryKingGrenade.java:37]: make link to `this(world)` constructor, remove pickupStatus

File: src/main/java/net/journey/entity/projectile/EntitySentryKingGrenade.java [hunk 2]
```diff
@@ -29,25 +30,74 @@ public boolean apply(@Nullable Entity e) {
...
 	}
 
 	public EntitySentryKingGrenade(World worldIn, EntityLivingBase shooter) {
 		super(worldIn, shooter);
 		this.pickupStatus = PickupStatus.DISALLOWED;
+		this.grenadeTimer = 200;
+
+	}
+	
+	public EntitySentryKingGrenade(World worldIn, EntityLivingBase shooter, int grenadeTime) {
```
Comment (TimeConqueror) [inline src/main/java/net/journey/entity/projectile/EntitySentryKingGrenade.java:51]: make link to this(world, shooter) constructor, remove pickupstatus

File: src/main/java/net/journey/entity/projectile/EntitySentryKingGrenade.java [hunk 3]
```diff
@@ -29,25 +30,74 @@ public boolean apply(@Nullable Entity e) {
...
 	}
 
 	public EntitySentryKingGrenade(World worldIn, double x, double y, double z) {
 		super(worldIn, x, y, z);
 		this.pickupStatus = PickupStatus.DISALLOWED;
+		this.grenadeTimer = 200;
+
+	}
+	
+	public EntitySentryKingGrenade(World worldIn, double x, double y, double z, int grenadeTime) {
```
Comment (TimeConqueror) [inline src/main/java/net/journey/entity/projectile/EntitySentryKingGrenade.java:65]: make link to `this(world, x,y,z)` constructor, remove pickupstatus

File: src/main/java/net/journey/entity/projectile/EntitySentryKingGrenade.java [hunk 4]
```diff
@@ -19,6 +19,7 @@
 import java.util.List;
 
 public class EntitySentryKingGrenade extends EntityTippedArrow implements IProjectile {
```
Comment (TimeConqueror) [inline src/main/java/net/journey/entity/projectile/EntitySentryKingGrenade.java:21]: Are all these constructors really needed?

File: src/main/java/net/journey/entity/projectile/EntitySentryKingGrenade.java [hunk 5]
```diff
@@ -29,25 +30,74 @@ public boolean apply(@Nullable Entity e) {
...
+			double d4 = d0 / d3;
+			double d5 = d2 / d3;
+			this.setLocationAndAngles(e.posX + d4, this.posY, e.posZ + d5, f2, f3);
+			float f4 = (float) (d3 * 0.20000000298023224D);
+			this.shoot(d0, d1 + f4, d2, f, f1);
+		}
+		this.pickupStatus = EntityArrow.PickupStatus.ALLOWED;
+	}
+	
+	public EntitySentryKingGrenade(World worldIn, EntityLivingBase e, EntityLivingBase eb, float f, float f1, int grenadeTime) {
```
Comment (TimeConqueror) [inline src/main/java/net/journey/entity/projectile/EntitySentryKingGrenade.java:96]: wtf is f, f1

File: src/main/java/net/journey/entity/projectile/EntitySentryKingGrenade.java [hunk 6]
```diff
@@ -29,25 +30,74 @@ public boolean apply(@Nullable Entity e) {
...
 
 	}
 
 	public EntitySentryKingGrenade(World worldIn, EntityLivingBase e, EntityLivingBase eb, float f, float f1) {
 		super(worldIn);
 		Entity.setRenderDistanceWeight(10.0D);
 		this.shootingEntity = e;
+		this.grenadeTimer = 200;
+
+		this.posY = e.posY + e.getEyeHeight() - 0.10000000149011612D;
```
Comment (TimeConqueror) [inline src/main/java/net/journey/entity/projectile/EntitySentryKingGrenade.java:78]: shooting in constructor is a bad way of doing things.
In constructor you initializes the object, without any actions

File: src/main/java/net/journey/entity/projectile/EntitySentryKingGrenade.java [hunk 7]
```diff
@@ -29,25 +30,74 @@ public boolean apply(@Nullable Entity e) {
...
+		double d0 = eb.posX - e.posX;
```
Comment (TimeConqueror) [inline src/main/java/net/journey/entity/projectile/EntitySentryKingGrenade.java:79]: *just marked for requesting changes*

**Review by TimeConqueror — CHANGES_REQUESTED** [review]

File: src/main/java/net/journey/entity/projectile/EntitySentryKingGrenade.java (hunk 4 above)
Comment (DiamondWalker) [reply src/main/java/net/journey/entity/projectile/EntitySentryKingGrenade.java:21]: Some of them could probably be removed. I just made variants for each existing constructor.

Comment (DiamondWalker) [reply src/main/java/net/journey/entity/projectile/EntitySentryKingGrenade.java:21]: Actually it seems removing them causes a crash.

**Pull request closed without merging**